packages/payment/src/workflows/steps/create-payment-provider-session.ts

```diff
@@ -0,0 +1,39 @@
+import { PaymentProviderContext } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  provider_id: string
+  data: {
+    context: PaymentProviderContext
+    amount: number
+    currency_code: string
+  }
+}
+
+export const createPaymentProviderSessionStepId =
+  "module:create-payment-provider-session"
+export const createPaymentProviderSessionStep = createStep(
+  createPaymentProviderSessionStepId,
+  async (input: StepInput, { container }) => {
+    const service = container.resolve("paymentProviderService")
+
+    const session = await service.createSession(input.provider_id, input.data)
```

Comment:
maybe we can better handle this in case of error to decide if we are throwing a regular error that can be retried by the workflow or if it's a permanent error.